src/main/java/org/aion/wallet/hardware/ledger/LedgerWallet.java

```diff
@@ -24,106 +30,73 @@
-    private static final String MAC_NPM_LOCATION = MAC_DRIVER_LOCATION + File.separator + "node-v8.11.4-darwin-x64/bin/";
-
-    private static final String LINUX_DRIVER_PATH = "native/linux/ledger/Aion-HID";
-    private static final String LINUX_PARAM_KEY = "--param=";
-
-    private static final int FIRST_INDEX = 0;
-    private static final byte PATH_LENGTH = 0x15;
-    private static final String AION_LEDGER_APP_PFX = "e0";
-    private static final String GET_PUBLIC_KEY_INDEX = "02";
-    private static final String SIGN_TRANSACTION_INDEX = "04";
-    private static final String DEFAULT_THIRD_AND_FORTH_BYTES_FOR_KEY_GENERATION = "0100";
-    private static final String DEFAULT_THIRD_AND_FORTH_BYTES_FOR_SIGNING = "0000";
-    private static final String DEFAULT_DERIVATION_PATH = "058000002c800001a98000000080000000";
-
-    private static final int HEX_KEY_SIZE = 64;
-
-    private static final WindowsNpmInstaller WINDOWS_NPM_INSTALLER = new WindowsNpmInstaller();
-    private static final String PREFIX = "--prefix";
 
     private static final int RETRIES = 3;
+    private static final int FIRST_INDEX = 0;
 
     private final Map<Integer, AionAccountDetails> accountCache = new ConcurrentHashMap<>();
-    private final ProcessBuilder processBuilder = createProcessBuilder();
 
     private final ExecutorService backgroundExecutor = Executors.newSingleThreadExecutor();
-
-    public LedgerWallet() {
-        installNpmIfRequired();
-    }
-
-    private ProcessBuilder createProcessBuilder() {
-        ProcessBuilder tmpBuilder = new ProcessBuilder();
-        OSUtils.executeForOs(this::updateProcessBuilderForWindows, this::updateProcessBuilderForMac, p -> {
-        }, tmpBuilder);
-        return tmpBuilder;
-    }
-
-    private void updateProcessBuilderForMac(final ProcessBuilder p) {
-        p.directory(new File(MAC_DRIVER_LOCATION));
-        final Map<String, String> environment = p.environment();
-        final String oldPath = environment.get("PATH");
-        environment.put("PATH", oldPath + File.pathSeparator + MAC_NPM_LOCATION);
-    }
-
-    private void updateProcessBuilderForWindows(final ProcessBuilder p) {
-        final Map<String, String> environment = p.environment();
-        final String oldPath = environment.get(PATH);
-        environment.put(PATH, oldPath + File.pathSeparator + WINDOWS_NPM_LOCATION);
-    }
-
-    private void installNpmIfRequired() {
-        OSUtils.executeForOs(WINDOWS_NPM_INSTALLER, p -> {}, p -> {}, processBuilder);
-    }
+    private ReentrantLock lock = new ReentrantLock();
 
     @Override
     public boolean isConnected() {
+        LedgerDevice ledgerDevice = null;
         try {
-            final AionAccountDetails firstAccount = getAccountDetails(FIRST_INDEX);
-            if (accountCache.containsKey(FIRST_INDEX) && !firstAccount.equals(accountCache.get(FIRST_INDEX))) {
-                accountCache.clear();
+            ledgerDevice = LedgerUtilities.findLedgerDevice();
```

Comment:
We can replace this with a call to `getAccountDetails(FIRST_INDEX);` which will remove the duplicated code and also ensure that we lock access to the Ledger resource